### PR TITLE
Fix Quick Access Glitches

### LINKED ISSUE #2835: Quick Access glitches
- Items are reversed in Quick Accesses. i.e, items opened recently appears at the bottom on the list.
- Same file appears multiple times.

### PR DESCRIPTION
**Pull request by VishnuSanal**
#### Issue tracker   
Fixes #2835 

#### Manual tests
- [x] Done

#### Build tasks success  
Successfully running following tasks on local:
- [x] `./gradlew assembledebug`
- [x] `./gradlew spotlessCheck`

### CODE REVIEW

**Comment by VishnuSanal** [comment]
Would adding a limit to the elements (say, 200) be a good idea?

File: app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/LoadFilesListTask.java [hunk 1]
```diff
@@ -480,7 +481,10 @@ else if (cursor.getCount() > 0 && cursor.moveToFirst()) {
     }
 
     UtilsHandler utilsHandler = AppConfig.getInstance().getUtilsHandler();
-    final LinkedList<String> paths = utilsHandler.getHistoryLinkedList();
+    LinkedList<String> historyLinkedList = utilsHandler.getHistoryLinkedList();
+    Collections.reverse(historyLinkedList);
+    LinkedHashSet<String> paths = new LinkedHashSet<>(historyLinkedList);
```
Comment (EmmanuelMess) [inline app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/LoadFilesListTask.java:486]: Why is it a HashSet now?

Comment (VishnuSanal) [reply app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/LoadFilesListTask.java:486]: > Why is it a HashSet now?

It doesn't allow duplicate items.

**Comment by VishalNehra** [comment]
This will move to 3.7

File: app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/LoadFilesListTask.java (hunk 1 above)
Comment (EmmanuelMess) [reply app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/LoadFilesListTask.java:486]: > > Why is it a HashSet now?
> 
> It doesn't allow duplicate items.

Why Linked?

File: app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/LoadFilesListTask.java [hunk 2]
```diff
@@ -480,7 +481,10 @@ else if (cursor.getCount() > 0 && cursor.moveToFirst()) {
...
+
     ArrayList<LayoutElementParcelable> songs = new ArrayList<>();
     for (String f : paths) {
```
Comment (EmmanuelMess) [inline app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/LoadFilesListTask.java:489]: Is this correct now?

**Comment by EmmanuelMess** [comment]
This should probably be created in the correct order, and uniquely, not fix it afterwards every time you open the view.

File: app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/LoadFilesListTask.java (hunk 2 above)
Comment (VishnuSanal) [reply app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/LoadFilesListTask.java:489]: > Is this correct now?

Order? Yes.

**Comment by VishnuSanal** [comment]
> This should probably be created in the correct order, and uniquely, not fix it afterwards every time you open the view.

I think it's created in the correct order. Because it's history & the first one goes in first. And, we show that in the history dialog too.

BTW, we should discuss on the feasibility of Quick Access, Recents & history. I created this PR to go with v3.6.2 before we decide on this puzzle. But, it didn't. So I think this PR isn't valid anymore.

**Pull request closed without merging**